packages/ui/components/avatar/avatar.test.tsx

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/* eslint-disable playwright/missing-playwright-await */
+import { render, screen, waitFor, act, fireEvent } from "@testing-library/react";
+import { useState } from "react";
+import { beforeAll, vi } from "vitest";
+
+import { Avatar, sizesPropsBySize } from "./Avatar";
+import { AvatarGroup } from "./AvatarGroup";
+
+type sizeKeys = "xxs" | "xs" | "xsm" | "sm" | "md" | "mdLg" | "lg" | "xl";
+
+vi.mock("@radix-ui/react-avatar", async () => {
+  const ImageMock = (props: { [key: string]: string; alt: string }) => {
+    return <img {...props} alt={props.alt} />;
+  };
+  const actual = (await vi.importActual("@radix-ui/react-avatar")) as { default: object };
+  return {
+    ...actual.default,
+    Image: ImageMock,
+  };
+});
+
```

Comment:
The Radix Image component needs to be mocked because it only renders after the image loads,
https://www.radix-ui.com/docs/primitives/components/avatar#image
and the image doesn't load because js-dom doesn't have some browser APIs (See below about ResizerObserve).